src/Redirection/DynamicRedirectionsSubscriber.php

```diff
@@ -40,12 +54,31 @@ public function onKernelException(GetResponseForExceptionEvent $event)
         $requestUri = rtrim($event->getRequest()->getRequestUri(), '/');
 
         /** @var Redirection $redirection */
-        $redirection = $this->repository->findOneBy(['from' => $requestUri]);
+        $redirection = $this->redirectRepository->findOneBy(['from' => $requestUri]);
 
-        if (!$redirection) {
-            return;
+        if ($redirection) {
```

Comment:
`if ($redirection = $this->redirectRepository->findOneBy(['from' => $requestUri])) {`